test/shelf_io_test.dart

```diff
@@ -221,7 +221,9 @@ void main() {
 
     var response = await _post(body: 'Hello');
     expect(response.statusCode, HttpStatus.notFound);
-    expect(response.headers['date'], 'Mon, 23 May 2005 22:38:34 GMT');
+    // TODO: after https://dart-review.googlesource.com/c/sdk/+/143081 lands,
```

Comment:
Need to be careful here. We test this against multiple SDKs.

Reply:
Does it work to make it lowercase on line 212 instead?

Reply:
> Need to be careful here. We test this against multiple SDKs.

OK. Any additional tests I need to run?
> Does it work to make it lowercase on line 212 instead?

No. Http parser will not enforce lowercased field and the key will be `Date`(The same as the input).

Reply:
It will match the case of what was given to it right? Right now we are giving a capital `Date` - if we give it a lowercase `date` would it keep it?

Reply:
Exactly. It will keep!

Reply:
Line 212 is where it is effectively passed in. If we make that lowercase and leave this line as is I think we'll be backwards and forwards compatible.

Reply:
Good point! That's even better. Are you able to modify the pr here or I have to create another pr?

Reply:
I updated the PR, you should be able to `git pull` to see it locally.

Can you run against your patched SDK to confirm, or do you want me to pull down the change locally?

Reply:
It worked with my change.

Reply:
Awesome. Thanks!